reload/ddsdk2016
# Upgrade search_api_solr and related modules DDSDK-498

#### What does this PR do?
Upgrades the search_api_solr, search_api, facets and search_api_entity_exclude modules.

The upgrade is complicated by two factors

1. We have created all the (danish) content on the site with langcode=en
2. We're running an old (6.x) version of solr that search_api_solr no longer has a configuration for.

I handled 1. by changing the stock search_api_solr configuration for english fields, to now be configured for danish. That is, the text_en solr field is now (amongst other things) configured with a danish dictionary. The alternative was to migrate all content over into danish instances of their fields which would be hard to do within the alotted time.

2. is handled by upgrading to solr 8.6.

Last but not least I've documented all of this in the sites documentation

#### Where should the reviewer start?
Read through the new documentation, the skim the commits and see if everything makes sense.

Relevant commits
* e0ef2d988b398bf99a01060d593786ea89adb184 standard composer upgrade of facets, search_api, search_api_solr and search_api_exclude_entity and a settings.php change as the module now requires a different path. Last but not least, a full config update - without modifications
* 710821095937ef3e7396b35113e8e7e2d13efd31 the modified solr configuration and and updated documentation that explains our approach. `configuration/drupal/search_api_solr.solr_field_type.text_en_7_0_0.yml` now contains the configuration that generates eg. our dictionary and stopwords list so it's quite big
* dcb8cc0a2e0cd9990dc186654de336bf7d2808b8 adds the re-generated _solr_ config used by platform.sh and our docker-setup. As a part of the upgrade I've had to upgrade to solr 8.6. I've also had to rename our platform sh service in `.platform/services.yaml` to ensure platform.sh does not attempt to boot solr on a 6.x core

#### How should this be manually tested?
Do a local reset and verify that everything comes up. Then index the site (`drush sapi-i` or manually via admin/config/search/search-api/index/nodes) and do a bunch of test searches.

I'll get the pr-site re-indexed and report back when it is ready for testing.

#### What are the relevant tickets?
DDSDK-498
